inst/markdown/nashConvergence-regiTarget.Rmd

```diff
@@ -0,0 +1,419 @@
+                                  "Deviation: ", round(100*.data$pm_emiMktTarget_dev_iter,2), "%", ifelse(pm_emiMktTarget_dev_iter >= 0, " > ", " < -"), round(100*emiMktTarget_tolerance[ext_regi],2),"%"))
+         )
+  )
+
+g[["deviation"]] <- 
+  lapply(setNames(tgtRegions, tgtRegions), function(reg){
+    subGroup <- setNames(tgts %>% filter(ext_regi == reg) %>% pull(ttot2), tgts %>% filter(ext_regi == reg) %>% pull(group))
+    lapply(subGroup, function(tgtPeriod){
+      d <- plotData %>% filter(ext_regi == reg, ttot2 == tgtPeriod)
+      p <- ggplot(d, aes_string(x = "iteration", y = "pm_emiMktTarget_dev_iter")) +
+        suppressWarnings(geom_rect(aes(xmin = .data$iteration - 0.5, xmax = .data$iteration + 0.5,ymin = emiMktTarget_tolerance[ext_regi] * -1,ymax = emiMktTarget_tolerance[ext_regi], fill = converged, text = tooltip), alpha = 0.2)) +
+        suppressWarnings(geom_line(aes_string(group = 1, text = "tooltip"))) +
+        scale_y_continuous(breaks = c(-5*emiMktTarget_tolerance[ext_regi], -emiMktTarget_tolerance[ext_regi], 0, emiMktTarget_tolerance[ext_regi], 5*emiMktTarget_tolerance[ext_regi]), labels = scales::percent_format(accuracy = .1)) +
+        scale_fill_manual(values = booleanColor) +
+        theme_bw() +
+        theme(strip.background = element_blank(),
+              axis.title.x=element_blank(),
+              legend.position = "none") + 
+        coord_cartesian(ylim=c(-5*emiMktTarget_tolerance[reg], +5*emiMktTarget_tolerance[reg])) +
+        ylab("Deviation (%)") +
+        scale_x_continuous(expand=c(0,0),breaks=1:max(d$iteration)) +
+        #xlim(0,max(d$iteration)+1) +
+        theme(plot.margin = margin(0,42,0,22, "pt")) # t,r,b,l
+        
+      plotly::ggplotly(p, tooltip = c("text"), height = 200) #, autosize = FALSE
+    })
+  })
+
+
+```
+
+
+
+## {.tabset}
+
+```{r, include=FALSE}
+
+# Creating sections
+out <- ""
+for(reg in tgtRegions){
+  out <- paste0(out,knitr::knit_expand(text=paste0("### ", reg, " {.tabset}\n\n")))
+  for(subGroup in tgts %>% filter(ext_regi == reg) %>% pull(group)){
+    
+    tgtPeriod <- strsplit(subGroup,"_",fixed=T)[[1]][1]
+    unit <- pm_emiMktTarget %>% filter(ext_regi == reg, ttot2 == tgtPeriod) %>% pull(emi_type_47)
+    
+    out <- paste0(out,knitr::knit_expand(text=paste0("#### ", subGroup, "\n\n")))
+    
+    out <- paste0(out,knitr::knit_expand(text=paste0("##### Emissions and Carbon Price (", unit, ")\n\n")))
+    out <- paste0(out,"```{r , results = 'asis', fig.height=10}\n\n")
+    out <- paste0(out,"g$qtty_and_price[['", reg, "']][['", subGroup, "']]\n\n")
+    out <- paste0(out,"```\n\n")
+    
+#    out <- paste0(out,knitr::knit_expand(text=paste0("##### Deviation per iteration\n\n")))
+    out <- paste0(out,"```{r , results = 'asis', fig.height=10}\n\n")
+    out <- paste0(out,"g$deviation[['", reg, "']][['", subGroup, "']]\n\n")
+    out <- paste0(out,"```\n\n")
+  }
+}
+
+cat(out,file="output.txt", append=FALSE)
```

Comment:
This currently writes to an output.txt file in the inst folder of the library. This is most likely not intended!?